docker-entrypoint.sh

```diff
@@ -1,9 +1,45 @@
 #!/bin/bash
 set -e
 
-if [ ! -e '/var/www/html/version.php' ]; then
-    tar cf - --one-file-system -C /usr/src/nextcloud . | tar xf -
+# version_greater A B returns whether A > B
+function version_greater() {
+	[[ "$(printf '%s\n' "$@" | sort -V | head -n 1)" != "$1" ]];
+}
+
+installed_version="0.0.0"
+if [ -f /var/www/html/version.php ]; then
+    installed_version=$(php -r 'require "/var/www/html/version.php"; echo "$OC_VersionString";')
+fi
+image_version=$(php -r 'require "/usr/src/nextcloud/version.php"; echo "$OC_VersionString";')
+
+if version_greater "$installed_version" "$image_version"; then
+    echo "Downgrade not supported"
+    exit 1
+fi
+
+if version_greater "$image_version" "$installed_version"; then
+    if version_greater "$installed_version" "0.0.0"; then
```

Comment:
Maybe we should do something like `[[ "$installed_version" != "0.0.0" ]]` in all of these cases to make it more clear which is the upgrade path and which is the fresh install path?